development.md

```diff
@@ -0,0 +1,91 @@
+* [Building](#building)
+* [Running tests](#running-tests)
+* [Writing and Generating Documentation](#writing-and-generating-documentation)
+* [Linting](#linting)
+* [Workflow](#workflow)
+* [Resources](#resources)
+
+## Things to install
+* [swiftenv](https://swiftenv.fuller.li/en/latest/installation.html): a command-line tool that allows easy installation of and switching between versions of Swift.
+	* Use this to install Swift 4.0 if you don’t have it already.
+* [jazzy](https://github.com/realm/jazzy#installation): the tool we use to generate documentation.
+* [swiftlint](https://github.com/realm/SwiftLint#using-homebrew): the Swift linter we use. 
+* [libmongoc](http://mongoc.org/libmongoc/current/api.html): the MongoDB C driver, which this library wraps. See the installation instructions provided in our [README](README.md#first-install-the-mongodb-c-driver) or on the [libmongoc docs](http://mongoc.org/libmongoc/current/installing.html). 
+
+## The code
+You should clone this repository, as well as the [MongoDB Driver specifications](https://github.com/mongodb/specifications). 
+Since this library wraps the MongoDB C Driver, we also recommend cloning [mongodb/mongo-c-driver](https://github.com/mongodb/mongo-c-driver) so you have the source code and documentation easily accessible. 
+
+## Building 
+### From the command line
+Run `make` from the project's root directory. 
+
+### In Xcode
+We do not provide or maintain an already-generated `.xcodeproj` in our repository. Instead, you must generate it locally.
+
+**To generate the `.xcodeproj` file**:
+1. Install the Ruby gem `xcodeproj` with `gem install xcodeproj` (you may need to `sudo`)
+2. Run `make project`
+3. You're ready to go! Open `MongoSwift.xcodeproj` and build and test as normal.
+
+Why is this necessary? The project requires a customized "copy resources" build phase to include various test `.json` files. By default, this phase is not included when you run `swift package generate-xcodeproj`. So `make project` first generates the project, and then uses `xcodeproj` to manually add the files to the appropriate targets (see `add_json_files.rb`). 
+
+## Running Tests
+**NOTE**: Several of the tests require a mongod instance to be running on the default host/port, `localhost:27017`.
+
+Additionally, please note that each benchmark test runs for a minimum of 1 minute and therefore **the entire benchmark suite will take around 20-30 minutes to complete**.
+
+You can run tests from Xcode as usual. If you prefer to test from the command line, keep reading.
+
+### From the command line 
+Tests can be run from the command line with `make test`. By default, this will run all the tests excluding the benchmarks.
+
+To only run particular tests, use the `FILTER` argument, which is passed as the `filter` argument to `swift test`. This will run test cases with names matching a regular expression, formatted as follows: `<test-target>.<test-case>` or `<test-target>.<test-case>/<test>`.
```

Comment:
> use the `FILTER` argument

Should this be "`FILTER` environment variable"? It's an argument/parameter to `swift test`, but in the example below it does look like an environment var.

Reply:
updated